gmake error: invalid application of ‘sizeof’ to incomplete type ‘KviMaskEntry’

On make or gmake the compile fails (see attached logs).

/home/jax5/temp/KVIrc-5.2.0/build/src/kvirc/kvirc_autogen/UYX5XTB5RZ/moc_KviChannelWindow.cpp:150:5:   required from here
/usr/include/x86_64-linux-gnu/qt6/QtCore/qmetatype.h:842:23: error: invalid application of ‘sizeof’ to incomplete type ‘KviMaskEntry’
  842 |         static_assert(sizeof(T), "Type argument of Q_PROPERTY or Q_DECLARE_METATYPE(T*) must be fully defined");
      |                       ^~~~~~~~~
gmake[2]: *** [src/kvirc/CMakeFiles/kvirc.dir/build.make:76: src/kvirc/CMakeFiles/kvirc.dir/kvirc_autogen/mocs_compilation.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:4981: src/kvirc/CMakeFiles/kvirc.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2

KVIrc:
KVIrc 5.2.0 tar.gz (grabbed the latest git and cloned it)

Debian:
Debian GNU/Linux 12.2.0-14

uname:
Linux main 6.1.0-13-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.55-1 (2023-09-29) x86_64 GNU/Linux

cmake:
cmake version 3.25.1

qmake:
QMake version 3.1
Using Qt version 6.4.2 in /usr/lib/x86_64-linux-gnu

gcc:
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/12/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:amdgcn-amdhsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Debian 12.2.0-14' --with-bugurl=file:///usr/share/doc/gcc-12/README.Bugs --enable-languages=c,ada,c++,go,d,fortran,objc,obj-c++,m2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-12 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --enable-libphobos-checking=release --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --enable-cet --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-12-bTRWOB/gcc-12-12.2.0/debian/tmp-nvptx/usr,amdgcn-amdhsa=/build/gcc-12-bTRWOB/gcc-12-12.2.0/debian/tmp-gcn/usr --enable-offload-defaulted --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 12.2.0 (Debian 12.2.0-14) 

Attached:
compile log
[compile.txt](https://github.com/kvirc/KVIrc/files/14071447/compile.txt)

--- a/src/kvirc/ui/KviChannelWindow.cpp
+++ b/src/kvirc/ui/KviChannelWindow.cpp
@@ -38,7 +38,6 @@
 #include "KviWindowListBase.h"
 #include "KviMainWindow.h"
 #include "KviConfigurationFile.h"
-#include "KviMaskEditor.h"
 #include "KviControlCodes.h"
 #include "KviModeEditor.h"
 #include "KviApplication.h"

--- a/src/kvirc/ui/KviChannelWindow.h
+++ b/src/kvirc/ui/KviChannelWindow.h
@@ -34,6 +34,7 @@
 #include "KviConsoleWindow.h"
 #include "KviWindow.h"
 #include "KviIrcUserDataBase.h"
+#include "KviMaskEditor.h"
 #include "KviPixmap.h"
 #include "KviUserListView.h"
 #include "KviTimeUtils.h"
@@ -55,13 +56,6 @@ class KviModeEditor;
 class KviTalHBox;
 class KviThemedLabel;
 class KviTopicWidget;
-
-#ifdef COMPILE_ON_WINDOWS
-// windows compiler wants this instead of the forward decl
-#include "KviMaskEditor.h"
-#else
-struct KviMaskEntry; // KviMaskEditor.h
-#endif
 
 /**
 * \def KVI_CHANNEL_AVERAGE_USERS The average channel users of a channel

